protocol/messenger_communities.go

```diff
@@ -2043,6 +2044,57 @@ func (m *Messenger) SetCommunityShard(request *requests.SetCommunityShard) (*Mes
 	return response, nil
 }
 
+func (m *Messenger) SetCommunityMailServers(request *requests.SetCommunityMailServers) (*MessengerResponse, error) {
+	if err := request.Validate(); err != nil {
+		return nil, err
+	}
+	community, err := m.communitiesManager.GetByID(request.CommunityID)
+	if err != nil {
+		return nil, err
+	}
+	if community == nil {
+		return nil, communities.ErrOrgNotFound
+	}
+	if !community.IsControlNode() {
+		return nil, errors.New("not admin or owner")
+	}
+
+	err = m.mailserversDatabase.SaveMailserversForCommunity(request.CommunityID, request.MailServers)
+	if err != nil {
+		return nil, err
+	}
+	if err := m.communityStorenodes.ReloadFromDB(); err != nil {
+		return nil, err
+	}
+	response := &MessengerResponse{
```

Comment:
Since we need to get these mailservers to the community members, we probably need a new message type that will be broadcasted when we save the storenodes